src/test/java/org/prebid/server/cache/CoreCacheServiceTest.java

```diff
@@ -839,6 +858,241 @@ public void cachePutObjectsShouldUseApiKeyWhenProvided() throws MalformedURLExce
+                .willReturn(new TextNode("VAST"))
+                .willReturn(new TextNode("updatedVast"));
+
+        // when
+        target.cachePutObjects(
+                asList(firstBidPutObject, secondBidPutObject, thirdBidPutObject),
+                true,
+                singleton("bidder1"),
+                "account",
+                "pbjs",
+                timeout);
+
+        // then
+        final BidPutObject modifiedFirstBidPutObject = firstBidPutObject.toBuilder()
+                .bidid(null)
+                .bidder(null)
+                .timestamp(null)
+                .key("account-high-entropy-")
+                .value(new TextNode("modifiedVast"))
+                .build();
+        final BidPutObject modifiedSecondBidPutObject = secondBidPutObject.toBuilder()
+                .bidid(null)
+                .bidder(null)
+                .timestamp(null)
+                .key("account-high-entropy-")
+                .build();
+        final BidPutObject modifiedThirdBidPutObject = thirdBidPutObject.toBuilder()
+                .bidid(null)
+                .bidder(null)
+                .timestamp(null)
+                .key("account-high-entropy-")
+                .value(new TextNode("updatedVast"))
+                .build();
+
+        assertThat(captureBidCacheRequest().getPuts())
+                .containsExactly(modifiedFirstBidPutObject, modifiedSecondBidPutObject, modifiedThirdBidPutObject);
+    }
+
+    @Test
+    public void cachePutObjectsShouldPrependTraceInfoWithDatacenterWhenEnabled() throws IOException {
+        // given
+        target = new CoreCacheService(
+                httpClient,
+                new URL("http://cache-service/cache"),
+                "http://cache-service-host/cache?uuid=",
+                100L,
+                "ApiKey",
+                false,
+                true,
+                "apacific",
+                vastModifier,
+                eventsService,
+                metrics,
+                clock,
+                idGenerator,
+                jacksonMapper);
+
+        given(idGenerator.generateId()).willReturn("high-entropy-cache-id");
+
+        final BidPutObject firstBidPutObject = BidPutObject.builder()
+                .type("json")
+                .bidid("bidId1")
+                .bidder("bidder1")
+                .timestamp(1L)
+                .value(new TextNode("vast"))
+                .build();
+        final BidPutObject secondBidPutObject = BidPutObject.builder()
+                .type("xml")
+                .bidid("bidId2")
+                .bidder("bidder2")
+                .timestamp(1L)
+                .value(new TextNode("VAST"))
+                .build();
+        final BidPutObject thirdBidPutObject = BidPutObject.builder()
+                .type("text")
+                .bidid("bidId3")
+                .bidder("bidder3")
+                .timestamp(1L)
+                .value(new TextNode("VAST"))
+                .build();
```

Comment:
No need to test three objects, one will be sufficient. Here you test cache id functionality and not accompanying logic. Please, leave single put object to simplify test. And, as always, please, check for similar occurences.